### ISSUE TITLE
Kalender: Eventliste Menü: Weiterleitungsseite

### ISSUE THREAD

**Issue by mapfei** [contao/core#8140]
Die eingestellte Weiterleitungsseite wird (auch) in der Online-Demo beim Aufruf des entsprechenden Links nicht aufgerufen.

**Comment by fritzmg** [comment]
You should provide exact reproduction steps, not only one sentence.

**Comment by mapfei** [comment]
Ich habe im Modul "Eventliste-Menü" eine Weiterleitungsseite definiert. Diese wird jedoch beim Aufruf des entsprechenden Links im Frontend nicht aufgerufen.

Beispiel: statt

veranstaltungen-archiv.html?year=2014

wird die Seite aufgerufen, auf dem das Modul eingebunden ist, jedoch mit dem entsprechenden Parameter ?year=2014.

**Comment by sonjaeh** [comment]
Kann diesen Fehler bestätigen. Siehe auch: https://github.com/contao/core/issues/7953

**Comment by johannes-bittner** [comment]
Kann ich ebenfalls bestätigen.

**Comment by MaryLou1** [comment]
Kann das auch bestätigen. Fehler wie von mapfei beschrieben.

**Comment by fritzmg** [comment]
Hm, it does work in the online demo though?

**Comment by fritzmg** [comment]
I found the bug.

The `ModuleEventMenu` generates its `jumpTo` URL in its `generate()` Method: [ModuleEventMenu.php#L43](https://github.com/contao/core/blob/3.5.6/system/modules/calendar/modules/ModuleEventMenu.php#L43) which is saved in the protected `$strUrl` variable of the `Events` class, which all Event modules inherit from.

However, immediately after `parent::generate()` is called of course, which executes `ModuleCalendar::generate()`. There, the `$strUrl` variable is overwritten with the current request string: [ModuleCalendar.php#L70](https://github.com/contao/core/blob/3.5.6/system/modules/calendar/modules/ModuleCalendar.php#L70).

This bug was introduced in [Contao 3.5.3](https://github.com/contao/core/blob/3.5.3/system/modules/calendar/modules/ModuleEventMenu.php#L43) - [previously](https://github.com/contao/core/blob/3.5.2/system/modules/calendar/modules/ModuleEventMenu.php) the `jumpTo` url was generated in each of the different `compile…` methods and thus `$this->strUrl` had the correct values.

As a workaround you could replace
```php
return parent::generate();
```
with
```php
$return = parent::generate();
$this->strUrl = preg_replace('/\?.*$/', '', \Environment::get('request'));
return $return;
```
in [ModuleEventMenu.php#L50](https://github.com/contao/core/blob/3.5.6/system/modules/calendar/modules/ModuleEventMenu.php#L50)

**Comment by fritzmg** [comment]
Sorry, had the wrong code in the workaround before, I have updated it now.

**Comment by weipah** [comment]
@fritzmg this workaround does not work for me. I'm still stuck when I go to the reader-site once, the URL in the event-menu still will try to redirect to itself.
Before the update to 3.6.5 it was in working condition.

Has anyone else tested the workaround so far?

**Comment by sonjaeh** [comment]
I just replaced the ModuleEventMenu.php with the previous (3.5.4) one, which at least worked for me.

**Comment by weipah** [comment]
@sonjaeh are you sure it was version 3.5.4?
Because there is no difference in the file ModuleEventMenu.php between these versions.

**Comment by fritzmg** [comment]
The one from `3.5.2` should work. The bug was introduced in `3.5.3`.

**Comment by weipah** [comment]
@fritzmg stimmt, hab es grad getestet. Mit der Datei aus dieser Version funktioniert es wieder. Danke dir!

**Comment by fritzmg** [comment]
@leofeyer shouldn't this be hotfixed? This bug is around since over 4 months now basically.

**Comment by leofeyer** [comment]
@fritzmg The ticket is already assigned to a bugfix branch.

**Comment by leofeyer** [comment]
Fixed in cc9a170f61f087fd29e0fdf7d3a6260be3442604.
